nav2_util/include/nav2_util/service_server.hpp

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2025 Maurice Alexander Purnawan
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_UTIL__SERVICE_SERVER_HPP_
+#define NAV2_UTIL__SERVICE_SERVER_HPP_
+
+#include <string>
+#include <memory>
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+
+namespace nav2_util
+{
+
+/**
+ * @class nav2_util::ServiceServer
+ * @brief A simple wrapper on ROS2 services for invoke() and block-style calling
+ */
+template<class ServiceT, typename NodeT = rclcpp::Node::SharedPtr>
+class ServiceServer
+{
+public:
+  using RequestType = typename ServiceT::Request;
+  using ResponseType = typename ServiceT::Response;
+  using CallbackType = std::function<void(const std::shared_ptr<rmw_request_id_t>,
+      const std::shared_ptr<RequestType>, std::shared_ptr<ResponseType>)>;
+
+  explicit ServiceServer(
+    const std::string & service_name,
+    std::string service_introspection_mode,
```

Comment:
This could/should be removed and done in the constructor as well using the `node` input